pytorch_lightning/trainer/trainer.py

```diff
@@ -1595,6 +1595,12 @@ def _call_callback_hooks(
                     with self.profiler.profile(f"[Callback]{callback.state_key}.{hook_name}"):
                         fn(self, self.lightning_module, *args, **kwargs)
 
+        # TODO: remove if block in v1.8
+        if hook_name == "on_pretrain_routine_start":
+            self.on_train_start(*args, **kwargs)
+        if hook_name == "on_pretrain_routine_end":
+            self.on_train_end(*args, **kwargs)
+
```

Comment:
This should not be done. We must support backwards compatibility for users who currently have implemented `on_pretrain_routine_start` or `on_pretrain_routine_end` . This breaks those use cases
```suggestion
```